app/features/quizzify/tools.py

```diff
@@ -25,6 +25,23 @@
 
 logger = setup_logger(__name__)
 
+def transform_json_dict(input_data: dict):
+    # Check if each choice has 'key' and 'value'
```

Comment:
I think you'll need a way to gracefully handle when the output is not in the expected format. Just so this function by itself can give proper handling

I believe we actually validate this earlier, but keeping downstream functions strong to stand on their own creates reusability in the codebase

Reply:
@mikhailocampo I understand. How can I improve that code?

Reply:
Yes, I saw that you used a Pydantic Schema. But I think that the main problem happens due to detecting the choices object as a list and not as a dict: `if isinstance(choices, dict):`
This is my result for `print(type(choices)) print(isinstance(choices, dict))`
![image](https://github.com/radicalxdev/kai-ai-backend/assets/113047749/8fd24f9e-2f81-4bc5-acb9-bc05f6a35c9a)

Reply:
Done! I've updated the transform_json_dict for handling the output in a proper way, considering our existent schemas.
![image](https://github.com/radicalxdev/kai-ai-backend/assets/113047749/a088dc4e-635c-4156-84cb-2a8fc2cd55fd)